pandas/io/iceberg.py

```diff
@@ -52,6 +56,10 @@ def read_iceberg(
     scan_properties : dict of {str: obj}, optional
         Additional Table properties as a dictionary of string key value pairs to use
         for this scan.
+    engine : str, optional
```

Comment:
Should the `read_*` and `to_*` signatures also have an `engine_kwargs: dict[str, Any] | None` argument to allow specific engine arguments to be passes per implementation?

Reply:
Very good point. In `read_parquet` we already have a `**kwargs` for engine specific arguments. In `map`, `apply`... it's a normal `engine_kwargs` since `**kwargs` is used in some cases for the udf keyword arguments. I think for IO readers/writers `**kwargs` as `read_parquet` does is fine.

I didn't want to add the engine to all connectors in this PR to keep it simpler, but I'm planning to follow up with another PR that adds it, and adds `**kwargs` for connectors where it's not there already. Surely happy to add both things here if you prefer, just thought it would make reviewing simpler to keep the implementation separate from all the changes to parameters.

Reply:
if engine-specific kwargs are needed, isn't that a good reason to use `engine.read_whatever(path, **kwargs)` instead of pd.read_[...]?